src/tablestore.rs

```diff
@@ -446,4 +503,79 @@ mod tests {
         let result = ts.write_sst(&wal_id, table2).await;
         assert!(matches!(result, Err(error::SlateDBError::Fenced)));
     }
+
+    #[tokio::test]
+    async fn test_tablestore_sst_and_partial_cache_hits() {
+        // Setup
+        let os = Arc::new(InMemory::new());
+        let format = SsTableFormat::new(20, 1, None);
+        let block_cache = Arc::new(BlockCache::new(1000));
+        let ts = Arc::new(TableStore::new(
+            os.clone(),
+            format,
+            Path::from("/root"),
+            Some(block_cache.clone()),
+        ));
+
+        // Create and write SST
+        let id = SsTableId::Compacted(Ulid::new());
+        let mut writer = ts.table_writer(id);
+        for i in 0..20 {
+            writer.add(&[i], Some(&[i])).await.unwrap();
+        }
+        let handle = writer.close().await.unwrap();
+
+        // Read index
+        let index = Arc::new(ts.read_index(&handle).await.unwrap());
+
+        // Test 1: SST hit
+        let sst_blocks = ts
+            .read_blocks_using_index(&handle, index.clone(), 0..20)
+            .await
+            .unwrap();
+        assert_eq!(sst_blocks.len(), 20);
```

Comment:
Since the cache code in table store is doing some delicate stuff, would you mind doing a slightly more exhaustive test? I'd like to verify the actual blocks match what's expected, rather than just the length.